### ISSUE TITLE
With HOMEBREW_CASK_OPTS="--require-sha", brew bundle exits 0 even if installation is not successful

### ISSUE THREAD

**Issue by weakish** [Homebrew/homebrew-bundle#1275]
Provided that `HOMEBREW_CASK_OPTS="--require-sha"`,
even if installation is not successful due to checksum not defined,
`brew bundle` still exists 0 and not reporting any error.

For example: (chromium cask defines its `sha256` checksum as `:no_check`)

```sh
export HOMEBREW_CASK_OPTS="--require-sha"
echo 'cask "chromium"' | brew bundle --file=-
```

From the output, it seems everything is okay and chromium is installed:

```
==> Downloading https://formulae.brew.sh/api/cask.jws.json

Installing chromium
Homebrew Bundle complete! 1 Brewfile dependency now installed.
```

Also, the status code is 0.

```sh
; echo $?
0
```

However, in fact chromium is not installed:

```sh
; brew info chromium | fgrep 'Not installed'
Not installed
``` 

Strangely, running `brew bundle` with the verbose option, brew exists 1 correctly:

```sh
; echo 'cask "chromium"' | brew bundle --file=- --verbose
Installing chromium
Installing chromium cask. It is not currently installed.
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-cask/2786e56
Already downloaded: /Users/kyo/Library/Caches/Homebrew/downloads/a03ec19b95fb538ee676d9dd29426ec36977b638c0865408ab7dda1030f0ac33--chromium.rb
Error: Cask 'chromium' does not have a sha256 checksum defined and was not installed.
This means you have the --require-sha option set, perhaps in your HOMEBREW_CASK_OPTS.
Installing chromium has failed!
Homebrew Bundle failed! 1 Brewfile dependency failed to install.

; echo $?
1
```

Homebrew version: 4.2.2-18-gdcd267b

**Comment by thejan2009** [comment]
I can report the same symptoms - `brew bundle` reports a successful install although one of the packages failed to install for completely unrelated reasons. Manual `brew install` fails properoy.

**Comment by ian-h-chamberlain** [comment]
Confirmed, after a `brew update` it seems to be failing as expected, without `--verbose`:

```console
$ echo 'cask "emacs"' | brew bundle install --file=-; echo $?
Installing emacs
==> Downloading https://emacsformacosx.com/emacs-builds/Emacs-29.1-1-universal.dmg
Already downloaded: /Users/ichamberlain/Library/Caches/Homebrew/downloads/bc7dcf972996173def825339282d91fd01e6a66a1448b5b37ec126d4500c561f--Emacs-29.1-1-universal.dmg
==> Installing Cask emacs
Error: It seems there is already an App at '/Applications/Emacs.app'.
==> Purging files for version 29.1-1 of Cask emacs
Installing emacs has failed!
Homebrew Bundle failed! 1 Brewfile dependency failed to install.
1
```

**Comment by weakish** [comment]
> Please verify

Verified this issue is fixed (tested on Homebrew 4.2.2-29-gb3751bc). 🎉

**Comment by MikeMcQuaid** [comment]
@ian-h-chamberlain @weakish thanks for reporting back, really helpful 🙇🏻
